src/Stratis.Features.FederatedPeg/TargetChain/WithdrawalTransactionBuilder.cs

```diff
@@ -71,7 +74,14 @@ public Transaction BuildWithdrawalTransaction(uint256 depositId, uint blockTime,
             }
             catch (Exception error)
             {
-                this.logger.LogError("Could not create transaction for deposit {0}: {1}", depositId, error.Message);
+                if (error is WalletException walletException && walletException.Message == FederationWalletTransactionBuilder.NoSpendableTransactionsMessage)
```

Comment:
Should the 'Not enough funds` case be handled in a similar manner?

Reply:
I think that the federation wallet having not enough funds is an _actual_ exception. It shouldn't happen, something would be really wrong.